tests/unit/classifier/LDA_unittest.cc

```diff
@@ -233,6 +203,32 @@ TEST(LDA, DISABLED_CheckProjection_SVD)
 	EXPECT_NEAR(+7.96084156, projection_SVD[9], epsilon);
 }
 
+// label type exception test
+TEST(LDA, num_classes_in_labels_exception)
+{
+	SGVector<float64_t> lab;
+	SGMatrix<float64_t> feat;
+
+	CDenseLabels* labels;
+	CDenseFeatures<float64_t>* features;
+
+	generate_test_data<float64_t>(lab, feat);
+	labels = new CDenseLabels();
+	lab[0]=2;
```

Comment:
this is very indirect and makes the code hard to read. Why not create some really really simple test data to make the point with the exception?

Reply:
I have simplified it a bit more now :+1: